### DISCUSSION TITLE
Could you fix this

### DISCUSSION THREAD

**Discussion by VikasPinak** [community/community#72945] [Programming Help]
### Body

function fibonacci(n)
    if n <= 0
        return []
    elseif n == 1
        return [0]
    else
        fib_series = [0, 1]
        while length(fib_series) <= n  
            push(fib_series, fib_series[end - 1] + fib_series[end])
        end
        return fib_series
    end
end

n = 10
result = fibonacci(n)
println("Fibonacci series of $n numbers: ", result)

Please Fix this 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by apartyagi** [comment]
function fibonacci(n)
    if n <= 0
        return []
    elseif n == 1
        return [0]
    else
        fib_series = [0, 1]
        while length(fib_series) < n
            push!(fib_series, fib_series[end - 1] + fib_series[end])
        end
        return fib_series
    end
end

n = 10  # Change the value of n to the number of Fibonacci numbers you want
result = fibonacci(n)
println("Fibonacci series of $n numbers: ", result)



check this code i test this is working fine

**Comment by faridnec** [comment]
```julia
function fibonacci(n)
    if n <= 0
        return []
    elseif n == 1
        return [0]
    else
        fib_series = [0, 1]
        while length(fib_series) <= n
            push!(fib_series, fib_series[end - 1] + fib_series[end])
        end
        return fib_series[1:n]  # Adjusted to return only the first n numbers
    end
end

n = 10
result = fibonacci(n)
println("Fibonacci series of $n numbers: ", result)
```
here `push!` is used instead of `push` to modify the array in place and I added a condition to return only the first `n` numbers in the series.
